core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -87,12 +86,14 @@ type TransmitChecker interface {
 // - a monotonic series of increasing nonces for eth_txes that can all eventually be confirmed if you retry enough times
 // - transition of eth_txes out of unstarted into either fatal_error or unconfirmed
 // - existence of a saved eth_tx_attempt
-type EthBroadcaster struct {
+//
+// The type HEAD represents the chain's native head datatype.
+type EthBroadcaster[HEAD any] struct {
```

Comment:
Where is this generic type `HEAD` used? Is it only to propagate to the `estimator`?

Reply:
Yes. Is there a better way to handle that without using this generic on the EthBroadcaster level?

Reply:
I didn't see one :shrug:

Reply:
One stylistic thing that could help limit some of the noise might be to use:
```suggestion
type EthBroadcaster[H Head] struct {
```
With `type Head any` defined and documented elsewhere.

Reply:
Does that help?
"Head any" or "H Head" are equally puzzling for the puzzled, no?

Generics are always hard to read and understand, imo. Can't really solve that i think.

Reply:
It objectively reduces the spam elsewhere through the files, and using single letters is standard, idiomatic Go. We absolutely can and should be working our hardest to make these APIs readable for users.

Reply:
Are you saying generally, wherever we use generics, instead of saying [XYZ any], we should say [X XYZ]?
I'm not against using idiomatic go, if this is an established best practice.

Reply:
`any` is perfectly fine to use given the right circumstances. I still don't fully understand what we are trying to capture in our case, but aliasing something in for `any` was just meant as a way to document your expectations clearly on a type.

Single letter generic types are the only examples I have seen.